pkg/controller/configmap/configmap_controller.go

```diff
@@ -0,0 +1,85 @@
+package configmap
+
+import (
+	"os"
+
+	ca "github.com/operator-framework/operator-marketplace/pkg/certificateauthority"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// Add creates a new ConfigMap Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new ReconcileConfigMap.
+func newReconciler(mgr manager.Manager) *ReconcileConfigMap {
+	return &ReconcileConfigMap{}
+}
+
+// add adds a new Controller to mgr with r as the ReconcileConfigMap.
+func add(mgr manager.Manager, r *ReconcileConfigMap) error {
+	// Create a new controller
+	c, err := controller.New("configmap-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource ConfigMap.
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, getPredicateFunctions())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getPredicateFunctions returns the predicate functions used to identify the configmap
+// that contains Certificate Authority information.
+// True should only be returned when the ConfigMap is updated by the cert-injector-controller.
+func getPredicateFunctions() predicate.Funcs {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			// If the ConfigMap is ever changed we should kick off an event.
+			if e.MetaOld.GetName() == ca.TrustedCaConfigMapName {
+				log.Infof("Update")
```

Comment:
I'm going to remove this in an upcoming PR update after the ongoing e2e-tests finish.